### PR TITLE
refactor: Use BlockFinder from SDK

### PR DESCRIPTION
**Pull request by pxrl**
This change migrates from the UMA BlockFinder to the Across-local variant and enables us to drop @uma/sdk as a direct dependency. We still inherit it from other packages, though.

### CODE REVIEW

File: src/utils/BlockUtils.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
-import { BlockFinder } from "@uma/financial-templates-lib";
-import { Block, getProvider, getRedis, isDefined, setRedisKey, shouldCache } from "./";
+import { utils as sdkUtils } from "@across-protocol/sdk-v2";
+import { getProvider, getRedis, isDefined, setRedisKey, shouldCache } from "./";
 
-const blockFinders: { [chainId: number]: BlockFinder<Block> } = {};
+type BlockFinder = sdkUtils.BlockFinder;
+const { BlockFinder } = sdkUtils;
```
Comment (pxrl) [inline src/utils/BlockUtils.ts:5]: It's annoying to have to do this by ts requires it when BlockFinder is used for a type annotation :\

File: src/utils/BlockUtils.ts [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
  * for this chain ID. Otherwise creates a new block finder and adds it to the cache.
  * @param chainId
  * @returns
  */
-export async function getBlockFinder(chainId: number): Promise<BlockFinder<Block>> {
+export async function getBlockFinder(chainId: number): Promise<BlockFinder> {
   if (!isDefined(blockFinders[chainId])) {
     const providerForChain = await getProvider(chainId);
-    blockFinders[chainId] = new BlockFinder<Block>(providerForChain.getBlock.bind(providerForChain), [], chainId);
+    blockFinders[chainId] = new BlockFinder(providerForChain);
```
Comment (james-a-morris) [inline src/utils/BlockUtils.ts:18]: +1 for never having to do `.bind( )` again lol

**Review by james-a-morris — APPROVED** [review]
LGTM - this is a straightforward refactor and I expect that any compilation/test errors would be caught by the linter/test clients

File: src/utils/BlockUtils.ts (hunk 1 above)
Comment (pxrl) [reply src/utils/BlockUtils.ts]: Ah, nice! That's much better. Thanks!

**Pull request merged**